### PR TITLE
[misc] Add 'az version' 

### PR DESCRIPTION
**Pull request by jiasli**
Inspired by #11425

Add preview command `az version show` to show the versions of Azure CLI modules and extensions in JSON format, so that the output can be passed to downstream command or script for easy parsing. 

```json
> az version show
This command is in preview. It may be changed/removed in a future release.
{
  "azure-cli": "2.0.78",
  "azure-cli-command-modules-nspkg": "2.0.3",
  "azure-cli-core": "2.0.78",
  "azure-cli-dev-tools": "0.1.1",
  "azure-cli-nspkg": "3.0.4",
  "azure-cli-telemetry": "1.0.4",
  "azure-cli-testsdk": "0.2.4",
  "extensions": {
    "application-insights": "0.1.1"
  }
}
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/resource/_help.py [hunk 1]
```diff
@@ -1307,3 +1307,9 @@
 type: group
 short-summary: Manage resource tags.
 """
+
+helps['version'] = """
+type: command
+short-summary: Show the versions of Azure CLI modules and extensions in JSON format
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/resource/_help.py]: The output format can be configured through `-o` option, right?

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/resource/_help.py]: Sure thing!

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/resource/_help.py]: Nice. Maybe replacing `JSON fomat` with `format configured by -o ` in the short-summary would be more clear.

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/resource/_help.py]: Or maybe:

> Show the versions of Azure CLI modules and extensions in JSON format by default or format configured by --output.

**Comment by Juliehzl** [comment]
one small suggestion here is that can we format the output when running with `-o table`

**Comment by Juliehzl** [comment]
can we add test for your command to check the output instance?

**Comment by jiasli** [comment]
https://github.com/Azure/azure-cli/pull/11680#issuecomment-570147481

Yes. Please see previous comments.

**Comment by jiasli** [comment]
https://github.com/Azure/azure-cli/pull/11680#issuecomment-570147764

Thanks for the suggestion. Added as requested.

**Comment by Juliehzl** [comment]
> [#11680 (comment)](https://github.com/Azure/azure-cli/pull/11680#issuecomment-570147481)
> 
> Yes. Please see previous comments.

but when I run `az version -o table`, extensions information is lost, is this expected?

**Comment by jiasli** [comment]
> > [#11680 (comment)](https://github.com/Azure/azure-cli/pull/11680#issuecomment-570147481)
> > Yes. Please see previous comments.
> 
> but when I run `az version -o table`, extensions information is lost, is this expected?

Yes. Because `extensions` is not the at the first layer of the json.

**Pull request merged**